### PR TITLE
APPS-1288 Update catch 503 response

### PR DESCRIPTION
**Pull request by Gvaihir**

### CODE REVIEW

File: api/src/main/scala/dx/api/DxApi.scala [hunk 1]
```diff
@@ -1089,12 +1092,15 @@ case class DxApi(version: String = "1.0.0", dxEnv: DXEnvironment = DxApi.default
     * @param dxfile the dx file to download
     * @param overwrite whether to overwrite an existing file - if false, an
     *                  exception is thrown if the target path already exists
+    * @param retryLimit max number of retries
+    * @param cliRunner max number of retries
```
Comment (commandlinegirl) [inline api/src/main/scala/dx/api/DxApi.scala]: It looks like a copy/paste error (?)

File: api/src/test/scala/dx/api/DxApiTest.scala [hunk 2]
```diff
@@ -159,6 +186,23 @@ class DxApiTest extends AnyFlatSpec with Matchers with BeforeAndAfterAll {
...
   }
 
+  it should "uploadFile and ignore 503 response when throttles" in {
+    val dest = s"${dxTestProject.id}:/${uploadPath}/throttled/"
+    val fileToUpload = files.keys.head
+    val dxUploadCmd =
+      s"""dx upload "${fileToUpload.toString}" --brief${dest}"""
+    val mockRunner = mock[CommandRunner]
+    val throttleMsg = "Too many inbound requests, throttling requests for user-my_user, code 503. " +
+      "Request Params=blah-blah, Request Other Params=unavailable. Waiting 10000 years before retry..."
```
Comment (commandlinegirl) [inline api/src/test/scala/dx/api/DxApiTest.scala:196]: :-)

**Review by commandlinegirl — COMMENTED** [review]
LGTM modulo a docstring update

File: api/src/main/scala/dx/api/DxApi.scala (hunk 1 above)
Comment (Gvaihir) [reply api/src/main/scala/dx/api/DxApi.scala]: fixed in [e5e0606](https://github.com/dnanexus/dxScala/pull/112/commits/e5e06067986c57573d0d27d95701f523761a645d)

**Review by commandlinegirl — APPROVED** [review]
LGTM, thanks!

**Pull request merged**